Dockerfile

```diff
@@ -1,15 +1,23 @@
 FROM php:7.1-apache
 
+ARG USE_C_PROTOBUF=false
+
 RUN apt-get update && apt-get install -y libxml2 zlib1g-dev git unzip
 
+# Install PHP extension(s) required for development
 RUN docker-php-ext-install bcmath
 
+# Install and configure Composer
 RUN curl -sS https://getcomposer.org/installer | php
 RUN mv composer.phar /usr/local/bin/composer
 
+# Install and configure the gRPC extension
 RUN pecl install grpc
 RUN echo 'extension=grpc.so' >> $PHP_INI_DIR/conf.d/grpc.ini
 
+# Install and configure the C implementation of Protobuf extension if needed
+RUN if [ "$USE_C_PROTOBUF" = "false" ]; then echo 'Use PHP implementation of Protobuf'; else echo 'Use C implementation of Protobuf'; pecl install protobuf; echo 'extension=protobuf.so' >> $PHP_INI_DIR/conf.d/protobuf.ini; fi
```

Comment:
Nit: "Using" instead of "Use" would be a bit easy to understand when logged as a text.

Reply:
Done